### DISCUSSION TITLE
There is a way to use socket.io client inside a chrome extension (manifest v3)

### DISCUSSION THREAD

**Discussion by Elior2739** [socketio/socket.io#4623] [Q&A]
Hi

I'm trying a while to use socket.io-client inside a chrome extension service worker.

I tried multiple ways to accomplish this:

1. Use `"type": "module"` inside manifest.json and try to import from the cdn
  ```
  import idk from "https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.6.0/socket.io.min.js"
  
  console.log(idk)
  ```
  
  But it didn't work and throwed an error: `Uncaught SyntaxError: The requested module 'https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.6.0/socket.io.min.js' does not provide an export named 'default'`
  
  Also tried:
  ```
  import { io } from "https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.6.0/socket.io.min.js"
  
  console.log(idk)
  ```
  But throw an error: `Uncaught SyntaxError: The requested module 'https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.6.0/socket.io.min.js' does not provide an export named 'io'`

2. Try to import from a local file, I download the file manually from `https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.6.0/socket.io.min.js` and tried to use
```
import { io } from "./socket.js"
```

which throwed an error: `An unknown error occurred when fetching the script.`

manifest.json:
```
{
  "manifest_version": 3,
  "name": "Extension",
  "version": "1.0",
  "description": "extension",
  "permissions": [
    "storage",
    "tabs",
    "activeTab",
    "scripting"
  ],
  "background": {
    "service_worker": "scripts/background.js",
    "type": "module"
  },
  "action": {
    "default_popup": "popup/default/index.html"
  }
}
```

I'm really don't know what to do and what I'm doing wrong, So that's is the reason why I asked it here.

**Comment by tobycm** [comment]
I think to use import, you will have to use the `esm` version of `socket.io`
Try change `socket.io.min.js` to `socket.io.esm.min.js`

**Reply by nirgn975** [reply]
This worked for me.

**Comment by mt-ks** [comment]
Same issue.

**Reply by Elior2739** [reply]
I forgot about this discussion, You can use it inside a chrome extension but it's more complicated.

You can use Webpack for the background scripts, I'll check today if I have the files of the extension that I worked on if yes I will send you for example

**Reply by BingeCode** [reply]
> I forgot about this discussion, You can use it inside a chrome extension but it's more complicated.
> 
> You can use Webpack for the background scripts, I'll check today if I have the files of the extension that I worked on if yes I will send you for example

Could you share your setup? I am struggling with the same and cannot get it to work. @Elior2739

**Reply by Elior2739** [reply]
Hi, Sorry for the late response. I'll try to send the project itself tomorrow

**Comment by nikhil478** [comment]
@Elior2739 could you please send it again i am facing same issue :)

**Reply by nikhil478** [reply]
@mehmetbeyHZ have you find any workaround

**Reply by tobycm** [reply]
@nikhil478 https://github.com/tobycloud/youtube-watch-together/blob/main/extension/background.js
this is how I did it

**Comment by Reza1290** [comment]
use watchify, or some bundle lib
